www/assets/js/lib/api.ts

```diff
@@ -12,6 +13,12 @@ export const search = async (params: SearchQueryParams) => {
     })
   })
 
-  const results = await response.json()
+  let results: any = {}
+  if (isApiSuccessful(response.status)) {
```

Comment:
I'm not 100% certain about this but I think that due to where this is being checked, certain scenarios don't result in an error being thrown.  I tested 3 scenarios here:

1.  CORS error - This was an easy one to test because all you have to do is point `SEARCH_API_URL` at RC or production and try and hit it in a normal browser session.  This resulted in an infinite loading animation and a CORS error in the console.
2. Domain unreachable - I intentionally redirected `discussions-rc.odl.mit.edu` in my `/etc/hosts` file to `localhost`, which would cause the request to fail.  This resulted in an infinite loading animation and an error in the console.
3. Intentionally set a blank body in the code - This was the only way I was able to actually see the error message.  I went into `api.ts` and intentionally commented out the body line, replacing it with a blank string.  This ensured the API would return a 500 which allowed me to see the error.

So, I think we should investigate how to make sure cases 1 & 2 are handled.

Reply:
Thank you for doing this testing @gumaerc 
Regarding the CORS issue, I thought that it might be something out of scope as its occurrence and fix are via the API which we are hitting, but we can try to handle it here.
So I'll look into these 1 and 2 use-cases.